src/bookmarks/service/hooks/useBookmarkAddHandler.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+  ClientBookmarkCategoryItem,
+  useGETCategoryListQuery,
+} from '@/bookmarks/api/bookmark';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+
+export type DisClosureType = 'PUBLIC' | 'PRIVATE' | 'FRIENDS';
+
+const useBookmarkAddHandler = () => {
+  // SERVER
+  const { data, isLoading } = useGETCategoryListQuery({
+    userId: '1',
+  });
+  const [categoryList, setCategoryList] = useState<
+    ClientBookmarkCategoryItem[] | undefined
+  >(data);
+  const [selectedCategory, setCategory] = useState('');
+  useEffect(() => {
+    setCategoryList(data);
+    data && setCategory(data[0].id);
+  }, [data]);
+
+  // INTERACTION
+  // 1. URL 입력
+  const [url, setUrl] = useState<string>('');
+  const [isValidateUrl, setValidateUrl] = useState(false);
+
+  const onChangeUrl = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setUrl(e.target.value);
+  };
+
+  // 1-1. URL VALIDATION
+  const urlRegex = useMemo(
+    () =>
+      '^((http|https)://)[-a-zA-Z0-9@:%._\\+~#?&//=]{2,256}\\.[a-z]{2,6}\\b([-a-zA-Z0-9@:%._\\+~#?&//=]*)$',
+    [],
+  );
+  useEffect(() => {
+    if (url.match(urlRegex)) {
+      setTimeout(() => {
+        setValidateUrl(true);
+      }, 500);
+    } else {
+      setValidateUrl(false);
+    }
+  }, [url, setValidateUrl]);
```

Comment:
```suggestion
const isValidateUrl = url.match(urlRegex) 
```
p3. 해당 url이 유효한지 여부를 상태값 말고 그냥 변수로 관리해도 되지 않을까용??
혹시 타이머때문에 useEffect를 사용해서 업데이트하시는건지 궁금합니답!! (타이머를 사용하시는 이유도 궁금해용!)

Reply:
아 이 친구가 원래 제가 만들 때 url validation을 서버를 거쳐서 확인을 하고
그 여부에 따라서 바뀌는걸 임시로 타이머로 처리 했는데
현재는 맞는 URL인지 아닌지 판단만 하면 될 것 같아서
수정해 보겠습니다 😃